JRaviLab/molevolvr2.0
# Adds draft backend API, db integration, db schema migration

This PR replaces the "dummy" API from the previous PR with one that does the following:
- interacts with the postgres server to store and retrieve analysis objects, served from the `analysis` endpoint
- implements other very WIP endpoints, e.g. `stats`, just for testing
- adds database schema migrations via [Atlas](https://atlasgo.io/)
- adds more verbose JSON serialization *(to get around Plumber just crashing when it encounters unserializable values)*

Other non-API-related changes:
- adds WIP test harness that allows tests to be located in subdirectories
- adds `shell` control command to `run_stack.sh` that starts a shell in the backend container
 
Things you might want to test:
- ensure that everything you need to create your `.env` file is in `.env.TEMPLATE`
- ensure that you can start the stack by running `./run_stack.sh`
- ensure you can see the API at http://localhost:9050
- try making a POST request to analyses, possibly via the Swagger UI at http://localhost:9050/__docs__/
- after the stack is up, try running `./run_stack.sh shell` in another window to get a shell into the backend
  - try running the test harness script from there, e.g. `./run_tests.sh`
